addons/website_sale/controllers/main.py

```diff
@@ -254,15 +254,14 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
                 parent_category_ids.append(current_category.parent_id.id)
                 current_category = current_category.parent_id
 
-        product_count = Product.search_count(domain)
+        search_product = Product.search(domain)
+        product_count = len(search_product)
         pager = request.website.pager(url=url, total=product_count, page=page, step=ppg, scope=7, url_args=post)
-        products = Product.search(domain, limit=ppg, offset=pager['offset'], order=self._get_search_order(post))
+        products = Product.search([["id", "in", search_product._ids]], limit=ppg, offset=pager['offset'], order=self._get_search_order(post))
 
         ProductAttribute = request.env['product.attribute']
         if products:
-            # get all products without limit
-            selected_products = Product.search(domain, limit=False)
-            attributes = ProductAttribute.search([('attribute_line_ids.product_tmpl_id', 'in', selected_products.ids)])
+            attributes = ProductAttribute.search([('attribute_line_ids.product_tmpl_id', 'in', search_product.ids)])
```

Comment:
Whats mitigated bewtenn don't do it if view in frontend is not enable, but it risk to break some custo, e.g. theme that disable the real view, but use a custom view to do the same.

Reply:
I do not understand what you mean, it is the fact that `selected_products` became `search_product` ?
Note that the new name `search_product` was inspired by the version 13 code.

Reply:
Most of he time, the feature "attributes filters" is not enable, and so it this one is computed for nothing.
Since we cannot really change this behavior in stable, we could, if this part take a lot of time to be executed, at least make it optional with an Ir.config.parameter.

KR